iModelCore/BeSQLite/BeSQLite.cpp

```diff
@@ -823,6 +823,505 @@ void     SnappyFromBlob::Finish() {m_blobIO.Close();}
+        return true;
+    };
+
+    auto findIndexes = [&]() {
+        Statement stmt;
+        rc = stmt.Prepare(db, SqlPrintfString("PRAGMA [%s].[index_list]([%s])", dbName.c_str(), tableName.c_str()));
+        if (rc != BE_SQLITE_OK)
+            return false;
+
+        while ((rc = stmt.Step()) == BE_SQLITE_ROW) {
+            MetaData::IndexInfo info;
+            info.name = stmt.GetValueText(1);
+            info.unique = stmt.GetValueInt(2) != 0;
+            info.origin = stmt.GetValueText(3);
+            info.partial = stmt.GetValueInt(4) != 0;
+            info.sql = getSql(info.name, "index");
+            if (!findIndexColumns(info))
+                return false;
+            tableInfo.indexes.push_back(info);
+        }
+        return true;
+    };
+
+    Statement stmt;
+    rc = stmt.Prepare(db, SqlPrintfString("PRAGMA [%s].[table_list]", dbName.c_str()));
+    if (rc != BE_SQLITE_OK)
+        return rc;
+
+    while ((rc = stmt.Step()) == BE_SQLITE_ROW) {
+        if (tableName.CompareToI(stmt.GetValueText(1)) == 0) {
+            tableInfo.schema = stmt.GetValueText(0);
+            tableInfo.name = stmt.GetValueText(1);
+            tableInfo.type = stmt.GetValueText(2);
+            tableInfo.nColumns = stmt.GetValueInt(3);
+            tableInfo.hasRowId = stmt.GetValueInt(4) != 0;
+            tableInfo.isStrict = stmt.GetValueInt(5) != 0;
+            tableInfo.sql = getSql(tableName, "table");
+            if (!findColumns())
+                return rc;
+
+            if (!findForeignKeys())
+                return rc;
+
+            if (!findTriggers())
+                return rc;
+
+            if (!findIndexes())
+                return rc;
+
+            break;
+        }
+    }
+    return rc == BE_SQLITE_DONE || rc == BE_SQLITE_OK  ? BE_SQLITE_OK : rc;
+}
+
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+DbResult MetaData::SchemaDiff(DbCR lhsDb, DbCR rhsDb, std::vector<Utf8String>& patches, bool allowDrop) {
+    auto defaultFilter = [](auto const& table) {
+        return table.name.StartsWith("sqlite_") || table.type != "table";
+    };
+    return SchemaDiff(lhsDb, rhsDb, defaultFilter, patches);
+}
+
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+DbResult MetaData::SchemaDiff(DbCR lhsDb, DbCR rhsDb, std::function<bool(MetaData::TableInfo const&)> excludeFilter, std::vector<Utf8String>& patches, bool allowDrop) {
+    DbResult rc;
+    auto lhsTables = MetaData::QueryTableNames(lhsDb, "main", rc);
+    if (rc != BE_SQLITE_OK)
+        return rc;
+
+    auto rhsTables = MetaData::QueryTableNames(rhsDb, "main", rc);
+    if (rc != BE_SQLITE_OK)
+        return rc;
+
+    // if (excludeFilter == nullptr) {

```

Comment:
Did you comment this out intentionally? I ask because later, you use `excludeFilter` without checking if it's empty. If it is empty, it will throw `bad_function_call`.

Reply:
fixed. It should not call function if it was not provided.